code/mob/living/critter.dm

```diff
@@ -267,6 +277,23 @@ ABSTRACT_TYPE(/mob/living/critter)
 		return Burn
 
 	// end convenience procs
+	was_harmed(var/mob/M as mob, var/obj/item/weapon = 0, var/special = 0, var/intent = null)
+		if (src.ai)
+			src._ai_patience_count--
+			src.ai.was_harmed(weapon,M)
+			if(src.is_hibernating)
+				if (src.registered_area)
+					src.registered_area.wake_critters(M)
+				else
+					src.wake_from_hibernation()
+			// We were harmed, and our ai wants to fight back. Also we don't have anything else really important going on
+			if (src.ai_retaliates && src.ai.enabled && length(src.ai.priority_tasks) <= 0 && src.should_critter_retaliate())
+				var/datum/aiTask/sequence/goalbased/retaliate/task_instance = src.ai.get_instance(/datum/aiTask/sequence/goalbased/retaliate, list(src.ai, src.ai.default_task))
+				task_instance.targetted_mob = M
+				task_instance.start_time = world.time
```

Comment:
```suggestion
				task_instance.start_time = TIME
```
never trust byond's internal time stuff

Reply:
pretty sure `TIME` gives you world timeofday, I'm only really interested in number of ticks, so I think `world.time` is the right call here unless it does some weirdass byond stuff I'm not aware of.

Reply:
But why do you only want the number of ticks, surely you'd want it to be lag adjusted? (`TIME` is still in ticks, just adjusted for server lag)

Reply:
oh, I did not realise this. I thought `TIME` was wall-clock. Will change.